addons/binding/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/handler/AirVisualNodeHandler.java

```diff
@@ -0,0 +1,271 @@
+import org.openhab.binding.airvisualnode.internal.config.AirVisualNodeConfig;
+import org.openhab.binding.airvisualnode.internal.json.NodeData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import jcifs.smb.NtlmPasswordAuthentication;
+import jcifs.smb.SmbFile;
+import jcifs.smb.SmbFileInputStream;
+
+/**
+ * The {@link AirVisualNodeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Victor Antonovich - Initial contribution
+ */
+public class AirVisualNodeHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(AirVisualNodeHandler.class);
+
+    public static final String NODE_JSON_FILE = "latest_config_measurements.json";
+
+    private final Gson gson;
+
+    private ScheduledFuture<?> pollFuture;
+
+    private long refreshInterval;
+
+    private String nodeAddress;
+
+    private String nodeUsername;
+
+    private String nodePassword;
+
+    private String nodeShareName;
+
+    private NodeData nodeData;
+
+    public AirVisualNodeHandler(@NonNull Thing thing) {
+        super(thing);
+        gson = new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).create();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing AirVisual Node handler");
+
+        AirVisualNodeConfig config = getConfigAs(AirVisualNodeConfig.class);
+
+        if (config.address == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Node address must be set");
+        }
+        this.nodeAddress = config.address;
+
+        this.nodeUsername = config.username;
+
+        if (config.password == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Node password must be set");
+        }
+        this.nodePassword = config.password;
+
+        this.nodeShareName = config.share;
+
+        this.refreshInterval = config.refresh * 1000L;
+
+        schedulePoll();
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Node poll scheduled");
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateChannel(channelUID.getId(), true);
+        } else {
+            logger.debug("Can not handle command '{}'", command);
+        }
+    }
+
+    @Override
+    public void handleRemoval() {
+        super.handleRemoval();
+        stopPoll();
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        stopPoll();
+    }
+
+    private synchronized void stopPoll() {
+        if (pollFuture != null && !pollFuture.isCancelled()) {
+            pollFuture.cancel(false);
+        }
+    }
+
+    private synchronized void schedulePoll() {
+        stopPoll();
```

Comment:
This is not needed as openHAB calls `dispose` before calling `initialize`. Thus the `stopPoll` is already been called.